torch/autograd/profiler.py

```diff
@@ -1213,9 +1294,21 @@ def append(s):
         if append_node_id:
             row_values.append(evt.node_id)
         if has_input_shapes:
-            row_values.append(str(evt.input_shapes)[:SHAPES_COLUMN_WIDTH])
+            row_values.append(str(evt.input_shapes)[:shapes_column_width])
+        if has_stack:
+            src_field = ""
+            if len(evt.stack) > 0:
+                src_field = evt.stack[0][:src_column_width]
+            row_values.append(src_field)
         append(row_format.format(*row_values))
 
+        if has_stack:
+            empty_headers = [""] * (len(headers) - 1)
+            for entry in evt.stack[1:MAX_STACK_ENTRY]:
```

Comment:
should it be whatever value is passed to group_by_n? or is it only used when there's no grouping?

also, why does it slice the first element?

Reply:
it slices the first element bc we already outputted it, this code outputs table row by row and we use multiple rows per stacktrace

Reply:
evt.stack here is either the original stack of the event, or (when we use group by group_by_n) an averaged event which has it's stack already limited to the first group_by_n
this MAX_STACK_ENTRY check is a sanity check to avoid very long profiler outputs

Reply:
evt.stack is limited to top group_by_n when we do group by (key_averages);
this MAX_STACK_ENTRY check is a sanity check to avoid very long profiler outputs